process.rb

```diff
@@ -1,39 +1,39 @@
+#!/usr/bin/env ruby
+
 require 'json'
 require 'logger'
 require 'filesize'
 require 'fileutils'
 require 'pp'
 require 'shellwords'
+require 'yaml'
+require 'pry'
 
-$EXCLUDED_DIRS = ['/deal.with.these', '/raw', '/woodworking', '/paw patrol', '/wild.kratts', '/word world'].freeze
 $PROCESS_FILE_TAGS = ['processme', 'processme1080', 'processme720', 'processmehw1080', 'processmehw720'].freeze
-$VIDEO_EXTENSIONS = ['.mkv', '.mp4', '.m4v', '.divx', '.mpg', '.ts'].freeze
+$config = YAML.load_file("config.yml")
 
-$logger = Logger.new('logs/process.log', 'daily')
-$total_space_reduction = 0
+FileUtils.mkdir_p($config["log_loc"])
+$logger = Logger.new(File.join($config["log_loc"], "process.log"), 'daily')
 
-@paths = [
-          '/Volumes/storage/videos/sports',
-          '/Volumes/storage/videos/movies',
-          '/Volumes/storage/videos/tv.shows/'
-        ]
+$total_space_reduction = 0
 
 def get_files (path)
   Dir[ File.join(path, '**', '*') ]
     .reject { |f| File.directory? f}
-    .reject { |f| $EXCLUDED_DIRS.any? { |p| File.dirname(f).downcase.include? p.downcase } }
+    .reject { |f| ($config["excluded_dirs"] || []).any? { |p| File.dirname(f).downcase.include? p.downcase } }
     .select { |f| $PROCESS_FILE_TAGS.any? { |p| File.basename(f).downcase.include? p.downcase } }
 end
 
 class TranscodableFile
-  def initialize(file)
+  def initialize(file, path)
```

Comment:
Why pass the path here? You can get the path from the file variable `File.dirname(file)`

Reply:
It's not the file's path it's the base path set in `squish_paths`.  So when it puts it in `processed` it removes what was specified in `squish_paths`.

Reply:
Basically it should put files exactly where it did before 🤞🏻 🙂